gf/include/alps/gf/mesh.hpp

```diff
@@ -779,6 +782,205 @@ namespace alps {
         ///Stream output operator, e.g. for printing to file
         std::ostream &operator<<(std::ostream &os, const legendre_mesh &M);
 
+        /**
+         * Numerical mesh (T = double or std::complex<double>)
+         */
+        template<class T, int k>
+        class numerical_mesh : public base_mesh {
+            double beta_;
+            int dim_;//dimension of the basis
+
+            statistics::statistics_type statistics_;
+
+            std::vector<piecewise_polynomial<T,k> > basis_functions_;
+
+            bool valid_;
+
+            void set_validity() {
+                valid_ = true;
+                valid_ = valid_ && dim_ > 0;
+                valid_ = valid_ && dim_ == basis_functions_.size();
+                valid_ = valid_ && beta_ >= 0.0;
+                valid_ = valid_ && (statistics_==statistics::FERMIONIC || statistics_==statistics::BOSONIC);
+
+                for (int l=0; l < basis_functions_.size()-1; ++l) {
+                    valid_ = valid_ && (basis_functions_[l].section_edges() == basis_functions_[l+1].section_edges());
+                }
+            }
+
+            void check_validity() const {
+                if (!valid_) {
+                    throw std::runtime_error("numerical mesh has not been properly constructed!");
+                }
+            }
+
+        public:
+            typedef generic_index<numerical_mesh> index_type;
+
+            numerical_mesh(gf::statistics::statistics_type statistics=statistics::FERMIONIC):
+                    beta_(0.0), dim_(0), statistics_(statistics), valid_(false) {}
+
+            numerical_mesh(double b,  const std::vector<piecewise_polynomial<T,k> >&basis_functions,
+                           gf::statistics::statistics_type statistics=statistics::FERMIONIC):
+                    beta_(b), dim_(basis_functions.size()), basis_functions_(basis_functions), statistics_(statistics), valid_(false) {
+                set_validity();
```

Comment:
> Hiroshi Shinaoka:
> I have cleaned up the code bit. More comments for Doxygen. 
> It throws in the constructor if the object has not been constructed properly.

@shinaoka : am i looking at the right code?
I do not see a `throw` in the constructor, somehow...
And the commit (95b1b5face01b0d5294555fadf87b6ca83d4e232) is said to be made 15 days ago...